api/app/GraphQL/Mutations/SyncPoolScreeningQuestions.php

```diff
@@ -0,0 +1,84 @@
+use App\Models\PoolSkill;
+use App\Models\ScreeningQuestion;
+use Exception;
+use Illuminate\Support\Facades\DB;
+
+final class SyncPoolScreeningQuestions
+{
+    /**
+     * @param  null  $_
+     * @param  array{}  $args
+     */
+    public function __invoke($_, array $args)
+    {
+        // Run everything in a transaction
+        DB::beginTransaction();
+        try {
+            $pool = Pool::find($args['poolId']);
+            $existingQuestions = ScreeningQuestion::where('pool_id', '=', $args['poolId'])->get();
+            $incomingQuestions = is_array($args['screeningQuestions']) ? $args['screeningQuestions'] : [];
+            $incomingAssessmentStep = $args['assessmentStep'];
+            $incomingQuestionIds = [];
+
+            // Create/update incoming questions based on the existence of an id
+            foreach ($incomingQuestions as $incomingQuestion) {
+                if (isset($incomingQuestion['id'])) {
+                    array_push($incomingQuestionIds, $incomingQuestion['id']);
+                    $questionToUpdate = $existingQuestions->find($incomingQuestion['id']);
+                    if ($questionToUpdate === null) {
+                        throw new Exception('Screening question with id "'.$incomingQuestion['id'].'" does not exist.');
+                    }
+
+                    $questionToUpdate->question = $incomingQuestion['question'];
+                    if (isset($incomingQuestion['sortOrder'])) {
+                        $questionToUpdate->sort_order = $incomingQuestion['sortOrder'];
+                    }
+
+                    $questionToUpdate->save();
+                } else {
+                    $pool->screeningQuestions()->Create([
+                        'question' => $incomingQuestion['question'],
+                        'sort_order' => isset($incomingQuestion['sortOrder']) ? $incomingQuestion['sortOrder'] : null,
+                    ]);
+                }
+            }
+
+            // Remove old questions which aren't included in the sync request
+            foreach ($existingQuestions as $existing) {
+                if (! in_array($existing->id, $incomingQuestionIds)) {
+                    $existing->forceDelete();
+                }
+            }
+
+            // Upsert assessment step
+            $assessmentStep = AssessmentStep::updateOrCreate(
+                ['pool_id' => $pool->id, 'type' => AssessmentStepType::SCREENING_QUESTIONS_AT_APPLICATION->name],
+                [...(isset($incomingAssessmentStep['title']) ? ['title' => $incomingAssessmentStep['title']] : [])]
+            );
+            if (is_array($incomingAssessmentStep['poolSkills'])) {
+                foreach ($incomingAssessmentStep['poolSkills'] as $skillID) {
+                    $skill = PoolSkill::find($skillID);
+                    if ($skill === null || $skill->pool_id !== $pool->id) {
+                        throw new Exception('PoolSkill with id "'.$skillID.'" does not exist for given pool.');
```

Comment:
Any i18n strings should be moved to `packages/i18n/src/messages/apiMessages.ts` so that they can be translated.